src/libsyntax/ext/expand.rs

```diff
@@ -565,7 +548,9 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
             _ => unreachable!(),
         };
 
-        attr::mark_used(&attr);
+        if let NonMacroAttr { mark_used: false } = *ext {} else {
```

Comment:
Could a small comment be added here as to why everything else other than this one case gets a `mark_used`?